app/controllers/photos_controller.rb

```diff
@@ -1,14 +1,15 @@
 class PhotosController < ApplicationController
   before_action :set_photo, only: %i[ show edit update destroy ]
   before_action :ensure_current_user_is_owner, only: [:destroy, :update, :edit]
-
+  before_action :ensure_user_is_authorized, only: [:show]
   # GET /photos or /photos.json
   def index
     @photos = Photo.all
   end
 
   # GET /photos/1 or /photos/1.json
   def show
+    authorize @photo
```

Comment:
Nice job! You can use a before action in each of your controllers to authorize an instance of photo or a photo class object